accounts/abi/abi.go

```diff
@@ -117,11 +118,66 @@ func (abi ABI) Pack(name string, args ...interface{}) ([]byte, error) {
 	return append(method.Id(), arguments...), nil
 }
 
+// toGoSliceType prses the input and casts it to the proper slice defined by the ABI
+// argument in T.
+func toGoSlice(i int, t Argument, output []byte) (interface{}, error) {
+	index := i * 32
+	if index+96 > len(output) {
+		return nil, fmt.Errorf("abi: cannot marshal in to go slice type: length insufficient %d require %d", len(output), index+64)
+	}
```

Comment:
I'll put a comment `+64` forgot to update.

This code needs to change to the following check + comment


```go
// The length of the output needs to be at the very least 64 for the following canonical representation
// of a ABI slice: [offset(32), length(32)]
if index+32 > len(output) { ... }
// code
length := /* size of array in elements */
if index+32+(length*32) > len(output) { ... }
```